defaults.go

```diff
@@ -185,6 +185,10 @@ func IsDomainName(s string) (labels int, ok bool) {
 
 	const lenmsg = 256
 
+	if s == "\n" {
```

Comment:
technically dns is 8-bit clean, and \n is sort of a valid character in a dns name, so this corner case must be dealt with in the zone parser and not here

Reply:
@miekg after some dig into the parser code, it seems impossible to deal this case in the zone parser.

Because all the rr depends on the `toAbsoluteName()` need to deal this case. But different rr has different rdata format.

For example,
- the CNAME has the rdata format of `example.org`.
- the MX has the rdata format of `50 mail`
- the SVR has the rdata format of `0 5 5060 sipserver.example.com.`
- the HTTPS has the rdata format of `HTTPS 1 . port=8002`

You cannot detect the bad target without knowing the rdata format.

Fortunately, all target in the rdata need to use the `toAbsoluteName()` method to uniform the domain. It's seems appropriate to add the check logic in the `toAbsoluteName()` method.

Reply:
> the fact nothing is specified until you see a comment token.

This is only true for the CNAME case. For the rr with rdate containing both target and other information like MX or SVR, it fails.

For example, in `bad.example.org. MX 10 ; bad mx`, the target is after the priority of 10. If we want to deal this case, it seems we need to add some logic in the `case zExpectRdata:` branch. However,  in the MX example, its rdata is not empty, but still invalid. The parser main loop cannot detect the invalid target because it does not know the rdata format.

If checking target is not accepted in the `toAbsoluteName()` method, it seems no choose by do the validation in the `parse()` method of every related rr separately. There are more that 30 different rr types need to be updated.

Reply:
`toAbsolute` is def. the wrong place, I also see you point about the number of RR types.

Reply:
If the `toAbsoluteName` is the wrong place, would you kindly like to make any suggestion on which part could the check logic be placed? Or each RR that requires domain data should check in their own `parse()` method?

Reply:
I think the latter, unless something more general could be find in the parser itself - but I doubt it, because it ( think it only lexes) doesn't have this information. Maybe something more generic can be found about the RRs having this problem, i.e. why does `A` not run into this if `CNAME` does